### PR TITLE
chore(plugins): make it easier to debug plugin errors

### PR DESCRIPTION
**Pull request by hypeJunction**
Make it easier to debug plugin errors by providing more context in
thrown exceptions

### CODE REVIEW

File: engine/classes/ElggPluginPackage.php [hunk 1]
```diff
@@ -184,7 +187,7 @@ private function validate() {
 			if (!is_readable($this->path . $file)) {
 				$have_req_files = false;
 				$this->errorMsg =
-					_elgg_services()->translator->translate('ElggPluginPackage:InvalidPlugin:MissingFile', [$file]);
+					elgg_echo('ElggPluginPackage:InvalidPlugin:MissingFile', [$file]);
```
Comment (jeabakker) [inline engine/classes/ElggPluginPackage.php]: Is it possible to keep the `elgg_echo` on the same line a `$this->errorMsg`? now that the line is shorter.....

File: engine/classes/ElggPluginPackage.php [hunk 2]
```diff
@@ -234,7 +237,7 @@ private function hasReadableConfigFile() {
 		}
 
 		$this->errorMsg =
-			_elgg_services()->translator->translate('ElggPluginPackage:InvalidPlugin:UnreadableConfig');
+			elgg_echo('ElggPluginPackage:InvalidPlugin:UnreadableConfig');
```
Comment (jeabakker) [inline engine/classes/ElggPluginPackage.php]: Is it possible to keep the `elgg_echo` on the same line a `$this->errorMsg`? now that the line is shorter.....

File: engine/classes/ElggPluginPackage.php [hunk 3]
```diff
@@ -251,7 +254,7 @@ private function isNamedCorrectly() {
 			$required_id = $manifest->getID();
 			if (!empty($required_id) && ($required_id !== $this->id)) {
 				$this->errorMsg =
-					_elgg_services()->translator->translate('ElggPluginPackage:InvalidPlugin:InvalidId', [$required_id]);
+					elgg_echo('ElggPluginPackage:InvalidPlugin:InvalidId', [$required_id]);
```
Comment (jeabakker) [inline engine/classes/ElggPluginPackage.php]: Is it possible to keep the `elgg_echo` on the same line a `$this->errorMsg`? now that the line is shorter.....

File: engine/classes/ElggPluginPackage.php [hunk 4]
```diff
@@ -286,7 +289,7 @@ private function hasSaneDependencies() {
 			// only valid provide types
 			if (!in_array($provide['type'], $this->providesSupportedTypes)) {
 				$this->errorMsg =
-					_elgg_services()->translator->translate('ElggPluginPackage:InvalidPlugin:InvalidProvides', [$provide['type']]);
+					elgg_echo('ElggPluginPackage:InvalidPlugin:InvalidProvides', [$provide['type']]);
```
Comment (jeabakker) [inline engine/classes/ElggPluginPackage.php]: Is it possible to keep the `elgg_echo` on the same line a `$this->errorMsg`? now that the line is shorter.....

File: engine/classes/ElggPluginPackage.php [hunk 5]
```diff
@@ -297,7 +300,7 @@ private function hasSaneDependencies() {
 				foreach (${$dep_type} as $dep) {
 					if (!in_array($dep['type'], $this->depsSupportedTypes)) {
 						$this->errorMsg =
-							_elgg_services()->translator->translate('ElggPluginPackage:InvalidPlugin:InvalidDependency', [$dep['type']]);
+							elgg_echo('ElggPluginPackage:InvalidPlugin:InvalidDependency', [$dep['type']]);
```
Comment (jeabakker) [inline engine/classes/ElggPluginPackage.php]: Is it possible to keep the `elgg_echo` on the same line a `$this->errorMsg`? now that the line is shorter.....

File: engine/classes/ElggPluginPackage.php [hunk 6]
```diff
@@ -308,7 +311,7 @@ private function hasSaneDependencies() {
 
 						if ($version_compare) {
 							$this->errorMsg =
-								_elgg_services()->translator->translate('ElggPluginPackage:InvalidPlugin:CircularDep',
+								elgg_echo('ElggPluginPackage:InvalidPlugin:CircularDep',
```
Comment (jeabakker) [inline engine/classes/ElggPluginPackage.php]: Is it possible to keep the `elgg_echo` on the same line a `$this->errorMsg`? now that the line is shorter.....

**Comment by hypeJunction** [comment]
Part of #12026 now

**Pull request closed without merging**